Update cStringIO imports

rahulporuri opened the pull request:
fixes #94

mdickinson commented on apptools/logger/agent/attachments.py:
```diff
@@ -58,7 +58,7 @@ def _attach_directory(self, dir):
 
         import zipfile
-        from cStringIO import StringIO
+        from six.moves import cStringIO as StringIO
```
Would the simpler `from six import StringIO` work here?

mdickinson replied:
Actually, looking at the place where this is used, do we need `BytesIO` rather than `StringIO`?

mdickinson replied:
Looks like we do. On Python 3, working in a directory that contains a `tox.ini` file, I see this:
```
>>> z = zipfile.ZipFile(StringIO(), "w")
>>> z.write("tox.ini")
Exception ignored in: <function ZipFile.__del__ at 0x1076fe7b8>
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/zipfile.py", line 1789, in __del__
    self.close()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/zipfile.py", line 1807, in close
    self._write_end_record()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/zipfile.py", line 1911, in _write_end_record
    self.fp.write(endrec)
TypeError: string argument expected, got 'bytes'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/zipfile.py", line 1743, in write
    with open(filename, "rb") as src, self.open(zinfo, 'w') as dest:
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/zipfile.py", line 1470, in open
    return self._open_to_write(zinfo, force_zip64=force_zip64)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/zipfile.py", line 1583, in _open_to_write
    self.fp.write(zinfo.FileHeader(zip64))
TypeError: string argument expected, got 'bytes'
```

mdickinson commented:
And this also shows that unfortunately this code isn't being exercised by unit tests. Does it look possible to write a regression test? (And one that would catch the accidental misuse of `StringIO` instead of `BytesIO` for the zip files?)

rahulporuri commented:
@mdickinson i was going to say that writing tests will take too long but it didn't really. can you take another look at the PR? I _think_ i've addressed all of the comments and i've added the regression tests (to atleast catch trivial bugs).

rahulporuri commented:
sigh. im not sure what's going on with the CI. all of the linux jobs failed with the following errors, errors which i saw earlier in #92 and didn't investigate - 

```
======================================================================

FAIL: Test if relative paths are set correctly.

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/enthought/apptools/apptools/persistence/tests/test_file_path.py", line 48, in test_relative

    self.assertEqual(f.rel_pth, expect)

AssertionError: 'tmpaEZMUG/data/t.vtk' != '../tmp/tmpaEZMUG/data/t.vtk'

----------------------------------------------------------------------
```

mdickinson commented on apptools/logger/agent/attachments.py:
```diff
@@ -58,20 +58,20 @@ def _attach_directory(self, dir):
 
         import zipfile
-        from cStringIO import StringIO
+        from six import BytesIO
```
Could probably use io.BytesIO here instead of six.BytesIO.

mdickinson commented on apptools/logger/agent/tests/test_attachments.py:
```diff
@@ -0,0 +1,65 @@
+        self.addCleanup(shutil.rmtree, self.tmpdir)
+
+        self.tmpfile = os.path.join(self.tmpdir, "dummy_file.txt")
+        with io.open(self.tmpfile, 'w') as filehandle:
```
Let's use an encoding here, just to follow good practices.

mdickinson commented on apptools/logger/agent/tests/test_attachments.py:
```diff
@@ -0,0 +1,65 @@
+        message = attachments.message
+        self.assertTrue(message.is_multipart())
+        payload = message.get_payload()
+        self.assertTrue(len(payload), 1)
```
`self.assertEqual`?

mdickinson commented on apptools/logger/agent/tests/test_attachments.py:
```diff
@@ -0,0 +1,65 @@
+        message = attachments.message
+        self.assertTrue(message.is_multipart())
+        payload = message.get_payload()
+        self.assertTrue(len(payload), 1)
```
self.assertEqual

mdickinson commented on apptools/logger/plugin/logger_service.py:
```diff
@@ -1,9 +1,11 @@
 import zipfile
 
+# Third-party library imports
+from six import BytesIO
```
Suggest using `io.BytesIO` instead of `six.BytesIO` here; then we don't need to change this when we drop Python 2 support.

mdickinson commented on apptools/logger/plugin/tests/test_logger_service.py:
```diff
@@ -0,0 +1,43 @@
+            msg = logger_service.create_email_message(
+                fromaddr='', toaddrs='', ccaddrs='', subject='', priority=''
+            )
+        self.assertIsNotNone(msg)
```
I'd drop this, unless there's some a priori reason to expect the msg to be `None`. This check is redundant with the `assertIsInstance` check below.

mdickinson commented on apptools/logger/plugin/tests/test_logger_service.py:
```diff
@@ -0,0 +1,43 @@
+                fromaddr='', toaddrs='', ccaddrs='', subject='', priority='',
+                include_userdata=True,
+            )
+        self.assertIsNotNone(msg)
```
Same comment as above.

mdickinson commented on apptools/logger/plugin/tests/test_logger_service.py:
```diff
@@ -0,0 +1,43 @@
+                include_userdata=True,
+            )
+        self.assertIsNotNone(msg)
+        self.assertIsInstance(msg, MIMEMultipart)
```
Missing trailing newline!
